CMakeLists.txt

```
@@ -13,13 +13,25 @@ set(CMAKE_ARCHIVE_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib)
 set(CMAKE_LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib)
 set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/bin)
 
-find_package(LLVM REQUIRED)
-message(STATUS "Found LLVM ${LLVM_PACKAGE_VERSION}")
-message(STATUS "Using LLVMConfig.cmake in ${LLVM_DIR}")
+# Allow adding this project as a external project to the LLVM build system using LLVM_EXTERNAL_PROJECTS
+if(CMAKE_SOURCE_DIR STREQUAL CMAKE_CURRENT_SOURCE_DIR)
```

Can we use `PROJECT_IS_TOP_LEVEL` instead?

```suggestion
if(PROJECT_IS_TOP_LEVEL)
```

---

I didn't know that existed i just tried to use what another project did.